app/models/report_filter.rb

```diff
@@ -5,38 +5,67 @@ class ReportFilter
     resolved
     account_id
     target_account_id
-    by_target_domain
+    search_type
+    search_term
     target_origin
   ).freeze
 
+  IGNORED_PARAMS = %i(resolved search_type search_term).freeze
+
   attr_reader :params
 
   def initialize(params)
     @params = params
   end
 
   def results
-    scope = Report.unresolved
+    scope = if params[:resolved] == '-1'
+              Report.unscoped
+            elsif params[:resolved] == '1'
+              Report.resolved
+            else
+              Report.unresolved
+            end
 
     params.each do |key, value|
-      scope = scope.merge scope_for(key, value)
+      next if IGNORED_PARAMS.include? key.to_sym
+
+      new_scope = scope_for(key, value)
+      scope = scope.merge new_scope if new_scope
     end
 
+    scope = scope.merge search_scope if searching?
     scope
   end
 
+  def search_filter
```

Comment:
We probably need input validation here, so checking that the input is either a well-formed domain, or a `@username` or `@username@domain.example`

Currently if you enter say `@username@123` it just returns empty results.

Reply:
I tried using the `AccountSearchService::MENTION_ONLY_RE` to raise an error, but that seemingly doesn't error on `@username@123`, but it will error on say `@username 123`